packages/cli/generators/repository/index.js

```diff
@@ -38,6 +61,37 @@ module.exports = class RepositoryGenerator extends ArtifactGenerator {
     super(args, opts);
   }
 
+  /**
+   * Find all the base artifacts in the given path whose type matches the
+   * provided artifactType.
+   * For example, a artifactType of "repository" will search the target path for
+   * matches to "*.repository.base.ts"
+   * @param {string} dir The target directory from which to load artifacts.
+   * @param {string} artifactType The artifact type (ex. "model", "repository")
+   */
+  async _findBaseClasses(dir, artifactType) {
+    const paths = await utils.findArtifactPaths(dir, artifactType + '.base');
+    debug(`repository artifact paths: ${paths}`);
+
+    // get base class and path
+    const baseRepositoryList = new Array();
+    _.forEach(paths, p => {
+      // get name removing anything from .artifactType.base
+      const firstWord = _.first(_.split(_.last(_.split(p, path.sep)), '.'));
+      const artifactName =
+        utils.toClassName(firstWord) + utils.toClassName(artifactType);
+
+      // get filename
+      const artifactFile = path.parse(_.last(_.split(p, path.sep))).name;
```

Comment:
> `_.last(_.split(p, path.sep)))`

This code is repeated twice (see `firstWord` on L80 above) and is also a bit difficult to understand when reading the code. Please extract it into a new variable with a descriptive name, e.g.

```ts
const fileNameWithExtension = _.last(_.split(p, path.sep)));
```

I think this entire "forEach" callback could be simplified by using `path.parse`, instead of ` _.last(_.split(p, path.sep)))`.

```ts
const artifactFile = path.parse(_.last(_.split(p, path.sep))).name;
const firstWord = _.first(artifactFile, _.split('.'));
// etc.
```